### PR TITLE
Update snsdemo commit

### PR DESCRIPTION
**Pull request by sa-github-api**
# Motivation
We would like to keep the testing environment, provided by snsdemo, up to date.

# Changes
* Updated `snsdemo` version in `dfx.json`.
* Ensured that the `dfx` version and IC commit in `dfx.json` match `snsdemo`.
* Updated the NNS canister `.did` file imports and derived rust code.
* TODO: The code will probably NOT compile if there have been significant changes to the NNS-dapp APIs.
  Please check the Rust code and make minimal changes to make the code compile.

# Tests
  - [ ] Please check the API updates for any breaking changes that affect our code.
  - [ ] Please check for new proposal types.

### CODE REVIEW

**Comment by bitdivine** [comment]
Note: I checked out the PR branch locally and resolved the imported rust code updates, keeping the rust code as close to upstream as possible, and the patch files as small as possible.  I would appreciate a review from someone else.

File: rs/sns_aggregator/src/types/ic_sns_ledger.patch [hunk 1]
```diff
@@ -1,76 +1,46 @@
 diff --git b/rs/sns_aggregator/src/types/ic_sns_ledger.rs a/rs/sns_aggregator/src/types/ic_sns_ledger.rs
```
Comment (bitdivine) [inline rs/sns_aggregator/src/types/ic_sns_ledger.patch:1]: The patch files exist to turn the auto-generated Rust files into something that complies; they should be as small as possible.

File: rs/sns_aggregator/src/types/ic_sns_ledger.rs [hunk 2]
```diff
@@ -32,6 +32,11 @@ pub enum ChangeFeeCollector {
     Unset,
```
Comment (bitdivine) [inline rs/sns_aggregator/src/types/ic_sns_ledger.rs:32]: The actual API changes!

**Review by mstrasinskis — APPROVED** [review]
Thanks, LGTM

**Comment by bitdivine** [comment]
I think the most efficient way of merging this to main is to cherry-pick the change across to main.  This way the changes are identical to those in this PR and unaffected by any tweaks to the automated updates.  If we have another automated update in say a week, any changes to the patch file are likely to be small.

File: declarations/sns_ledger/sns_ledger.did [hunk 3]
```diff
@@ -45,15 +45,21 @@ type MetadataValue = variant {
     Blob : blob;
 };
```
[the remark this answers is not in the document]
Comment (dskloetd) [reply declarations/sns_ledger/sns_ledger.did:47]: What do you mean by upstream? This file does not originate in snsdemo, does it?

Do you mean that we already made changes to the file when copying it into snsdemo and here we are using those same changes? Or is this file identical to a file in the IC repo?

[the remark this answers is not in the document]
Comment (bitdivine) [reply declarations/sns_ledger/sns_ledger.did:47]: This one:

> Or is this file identical to a file in the IC repo

File: dfx.json [hunk 4]
```diff
@@ -638,8 +638,8 @@
         "IDL2JSON_VERSION": "0.8.8",
         "OPTIMIZER_VERSION": "0.3.6",
         "DIDC_VERSION": "2023-07-25",
-        "SNSDEMO_RELEASE": "release-2023-07-07",
-        "IC_COMMIT": "06f339b83ce37e3fc9571e1b4251fbcf5c1a8239"
+        "SNSDEMO_RELEASE": "release-2023-09-07",
+        "IC_COMMIT": "6223a38cfae726e9cc83db9ae27f35ca979dd0d8"
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply dfx.json:642]: The snsdemo update contains an IC update, not the other way around, so the title is correct.

**Pull request closed without merging**